jax/_src/dtypes.py

```diff
@@ -458,18 +458,22 @@ def isdtype(dtype: DTypeLike, kind: str | DType | tuple[str | DType, ...]) -> bo
     True or False
   """
   the_dtype = np.dtype(dtype)
-  kind_tuple: tuple[DType | str, ...] = kind if isinstance(kind, tuple) else (kind,)
-  options: set[DType] = set()
+  kind_tuple: tuple[DTypeLike, ...] = kind if isinstance(kind, tuple) else (kind,)
+  options: set[DTypeLike] = set()
   for kind in kind_tuple:
-    if isinstance(kind, str):
-      if kind not in _dtype_kinds:
-        raise ValueError(f"Unrecognized {kind=} expected one of {list(_dtype_kinds.keys())}")
+    if isinstance(kind, str) and kind in _dtype_kinds:
       options.update(_dtype_kinds[kind])
-    elif isinstance(kind, np.dtype):
-      options.add(kind)
-    else:
-      # TODO(jakevdp): should we handle scalar types or ScalarMeta here?
-      raise TypeError(f"Expected kind to be a dtype, string, or tuple; got {kind=}")
+      continue
+    try:
+      _dtype = np.dtype(kind)
+      options.update([_dtype])
```

Comment:
`options.add(_dtype)` is more direct

Also, this should be outside the `try...except` statement. Why? What if `options.add` raises a TypeError? Then the error below would be misleading.